packages/search/src/search.tsx

```diff
@@ -156,35 +148,33 @@ const InnerSearch = (
 		hideClose,
 	}: Props,
 	forwardedRef: Ref< ImperativeHandle >
-): JSX.Element => {
+) => {
 	const { __ } = useI18n();
 	const [ keyword, setKeyword ] = React.useState( defaultValue );
 	const [ isOpen, setIsOpen ] = React.useState( defaultIsOpen );
 	const [ hasFocus, setHasFocus ] = React.useState( autoFocus );
 
 	const instanceId = useInstanceId( InnerSearch, 'search' );
-	const searchInput: MutableRefObject< HTMLInputElement | null > = React.useRef( null );
-	const openIcon: RefObject< HTMLButtonElement | null > = React.useRef( null );
-	const overlay: RefObject< HTMLDivElement | null > = React.useRef( null );
-
-	const focus = React.useCallback( () => {
-		// if we call focus before the element has been entirely synced up with the DOM, we stand a decent chance of
-		// causing the browser to scroll somewhere odd. Instead, defer the focus until a future turn of the event loop.
-		setTimeout( () => searchInput.current?.focus(), 0 );
-	}, [] );
-
-	const blur = React.useCallback( () => searchInput.current?.blur(), [] );
-
-	const clear = React.useCallback( (): void => setKeyword( '' ), [] );
+	const searchInput = React.useRef< HTMLInputElement >( null );
+	const openIcon = React.useRef< HTMLButtonElement >( null );
+	const overlay = React.useRef< HTMLDivElement >( null );
 
 	React.useImperativeHandle(
 		forwardedRef,
 		() => ( {
-			focus,
-			blur,
-			clear,
+			// if we call focus before the element has been entirely synced up with the DOM, we stand a decent chance of
+			// causing the browser to scroll somewhere odd. Instead, defer the focus until a future turn of the event loop.
+			focus() {
```

Comment:
Seems like declaring `focus` here will cause unexpected results - we're expecting that it's declared because we're calling it [here](https://github.com/Automattic/wp-calypso/pull/51059/files#diff-714aa6bf4174d2a19dc85f63e66b1cce1d33738c6b84f907fc66f8f7962dfb32R215).

Reply:
Good catch! It's weird that ESLint is not complaining about undeclared identifier. I'll check if this particular call in `openSearch` needs the timeout or if we can call `searchInput.current.focus()` directly.

Reply:
I guess it doesn't complain because  it expects that you meant `window.focus()` 😉

Reply:
I see. For JavaScript and ESLint, we disabled these `window.*` globals (there are hundreds of them) in #37953. We'll need some TypeScript solution for the same problem.

Related TS issue: https://github.com/microsoft/TypeScript/issues/14306

Reply:
I did a bit of research about why we need the `setTimeout` wrapper on the `focus` call and figured out we can remove it.

It was originally added by @blowery in #7883 to solve a bug with `<Search autoFocus />` inside a `<Popover />`. At that time, autofocus was implemented by calling `.focus()` inside `componentDidMount`. And at time of mount, the `Popover` wasn't done positioning yet -- it started measuring elements and figuring out its position also in `componentDidMount`. So we needed to delay the `.focus()` call a little bit.

Today, we merely pass the `autoFocus` prop to the inner native `<input>` element and the buggy `.focus()` calls no longer happen. So I removed the `setTimeout` in 917e843, simplifying the code nicely.